CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+
+Welcome! As a [Jupyter](https://jupyter.org) project, you can follow the [Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributing/content-contributor.html).
+
+Make sure to also follow [Project Jupyter's Code of Conduct](https://github.com/jupyter/governance/blob/master/conduct/code_of_conduct.md) for a friendly and welcoming collaborative environment.
+
+## Setting up a development environment
+
+LTIAuthenticator requires Python >= 3.6.
+
+1. Clone the repo:
+
+   ```bash
+   git clone https://github.com/jupyterhub/jupyterhub
+   ```
+
+2. Do a development install with `pip`:
+
+   ```bash
+   cd ltiauthenticator
+   python3 -m pip install --editable .
+   ```
+
+3. Install the development requirements, which include things like linting and testing tools:
+
+   ```bash
+   python3 -m pip install -r dev-requirements.txt
+   ```
+
+4. Set up pre-commit hooks for automatic code formatting and linting:
+
+   ```bash
+   pre-commit install
+   ```
+
+   You can also invoke the pre-commit hook manually at any time with:
+
+   ```bash
+   pre-commit run
+   ```
+
+## Contributing
+
+The LTIAuthenticator has adopted automatic code formatting so you shouldn't need to worry too much about your code style.
+As long as your code is valid, the pre-commit hook should take care of how it should look. You can invoke the pre-commit hook by hand at any time with:
+
+```bash
+pre-commit run
+```
+
+which should run any autoformatting on your code and tell you about any errors it couldn't fix automatically. You may also install [black integration](https://github.com/psf/black#editor-integration) into your text editor to format code automatically.
+
+If you have already committed files before setting up the pre-commit hook with `pre-commit install`, you can fix everything up using `pre-commit run --all-files`. You need to make the fixing commit yourself after that.
+
+## Testing
+
+It's a good idea to write tests to exercise any new features, or that trigger any bugs that you have fixed to catch regressions.
+
+From the root of the repo, you can run the tests with:
+
+```bash
+pytest -v
+```
+
+If you want to just run certain tests, check out the [pytest docs](https://pytest.readthedocs.io/en/latest/usage.html)
+for how pytest can be called. For example:
+
+```bash
+pytest -v -k spawn ltiauthenticator/tests/test_lti.py
+```
+
+## Commit Messages, Changelog, and Releases
+
+This project includes `pre-commit` hooks configured to run `black`, `flake8`, and `pip-compile` before running and completing the `git commit ...` command. If these tools do not run when running the `git add ...` and `git commit ...` commands, make sure you install the pre-commit hooks with `pre-commit install`.
+
+### For Contributors
+
+This project uses Semantic Versioning with Conventional Commits to track major, minor, and patch releases. The `npm run release` command automates [CHANGLOG.md](./CHANGELOG.md) updates and release version metadata.
```

Comment:
I don't think Jupyter projects in general make a commitment to semantic versioning, so let's keep that out for now?

Reply:
Done.